usr/share/rear/layout/prepare/default/319_rename_xfs_configs.sh

```diff
@@ -0,0 +1,79 @@
+# Cleanup directory which hold XFS configuration file for `rear recover'.
+# This will avoid possible mess in LAYOUT_XFS_OPT_DIR_RESTORE if `rear recover'
+# would be launched multiple times, where user will choose different disk
+# mapping each time.
+# Removing and creating LAYOUT_XFS_OPT_DIR_RESTORE will ensure that ReaR will
+# have only current files available during current session.
+rm -rf "$LAYOUT_XFS_OPT_DIR_RESTORE"
+mkdir -p "$LAYOUT_XFS_OPT_DIR_RESTORE"
+
+local excluded_configs=()
+
+# Read $MAPPING_FILE (disk_mappings) to discover final disk mapping.
+# Once mapping is known, configuration files can be renamed.
+# (e.g. sds2.xfs to sdb2.xfs, ...)
+while read source target junk ; do
+    # Disks in MAPPING_FILE are listed with full device path. Since XFS config
+    # files are created in format e.g. sda2.xfs strip prefixed path to have
+    # only short device name available.
+    base_source=$(basename "$source")
+    base_target=$(basename "$target")
+
+    # Check if XFS configuration file for whole device (unpartitioned)
+    # is available (sda, sdb, ...). If so, rename and copy it to
+    # LAYOUT_XFS_OPT_DIR_RESTORE.
+    if [ -e "$LAYOUT_XFS_OPT_DIR/$base_source.xfs" ]; then
+        Log "Migrating XFS configuration file $base_source.xfs to $base_target.xfs"
+        cp "$v" "$LAYOUT_XFS_OPT_DIR/$base_source.xfs" \
+         "$LAYOUT_XFS_OPT_DIR_RESTORE/$base_target.xfs"
+
+        # Replace old disk name in XFS configuration file as well.
+        sed -i s#"$base_source"#"$base_target"# \
```

Comment:
As far as I see that sed substitue changes wrong disks
when there are many disks e.g. from /dev/sda up to /dev/sdaa
and only /dev/sda should be mapped to /dev/sdb:
```
# echo sda >/tmp/test
# echo sdaa >>/tmp/test

# cat /tmp/test
sda
sdaa

# base_source=sda
# base_target=sdb

# ( set -x ; sed -i s#"$base_source"#"$base_target"# /tmp/test )
+ sed -i s#sda#sdb# /tmp/test

# cat /tmp/test
sdb
sdba
```